### PR TITLE
SAGEStack added

### PR DESCRIPTION
**Pull request by allaffa**
This is one of the simplest convolutional layers.
The introduction of this additional convolutional layer is minimally invasive with respect to the existing version of the repository, and helps us move towards the goal of introducing hyper parameter optimization in the repo, where the choice of the convolutional layer (GIN, GAT, PNA, CGCNN, SAGE) will be an hyper parameter to optimize to maximize validation accuracy.

### CODE REVIEW

**Comment by streeve** [comment]
Waiting on Pei's review? Or is this ready to merge?

**Comment by allaffa** [comment]
> Waiting on Pei's review? Or is this ready to merge?

This PR is standalone. It just adds a model to the list of models we currently support. We can wait. The most compelling PR for me is PR#105 because it always causes conflicts when I update it from the main branch

File: hydragnn/models/SAGEStack.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            dropout,
+            num_conv_layers,
+            num_nodes,
+        )
+
+    def get_conv(self, input_dim, output_dim):
+        return SAGEConv(
+            in_channels=input_dim,
+            out_channels=output_dim,
+            root_weight=False,
```
Comment (pzhanggit) [inline hydragnn/models/SAGEStack.py:55]: Any reason to  set `root_weight=False` and overwrite the default model formula?

Comment (allaffa) [reply hydragnn/models/SAGEStack.py:55]: Yes. I have not been completely honest with you guys. There are other hidden branches behind this PR that seems standalone XD
I am deliberately avoiding any **non-linearity** whenever I can to try to build a neural network that predicts properties that scale linearly with the size of the lattice.
I can introduce an additional input parameter to make this optional.

Comment (pzhanggit) [reply hydragnn/models/SAGEStack.py:55]: I hear you, but if I understand correctly, the`root_weight` here is not about square root of any weights. It is about the first part of 
image

Comment (allaffa) [reply hydragnn/models/SAGEStack.py:55]: You are right. I misinterpreted the description. I just looked at the implementation 
[https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/sage_conv.html#SAGEConv](url)
Screen Shot 2022-03-08 at 11 14 09 PM

File: tests/test_graphs.py [hunk 2]
```diff
@@ -122,6 +122,7 @@ def unittest_train_model(model_type, ci_input, use_lengths, overwrite_data=False
 
     # Set RMSE and sample MAE/max error thresholds
     thresholds = {
+        "SAGE": [0.45, 0.45, 0.90],
```
Comment (pzhanggit) [inline tests/test_graphs.py:125]: It seems SAGE is doing okay in the unit test after the changes. Maybe we can use the same error threshold as the other models?

**Review by pzhanggit — APPROVED** [review]

**Comment by allaffa** [comment]
Now I am afraid of merging anything before PR#105. XD
